### PR TITLE
[NO GBP] Fixes chair, echair, wheelchair and vehicle overlays on painted objects

### LINKED ISSUE #88644: Wheelchair wheel underlay is below the human mob sprite.
### Client Version:

515.1647

### Issue Summary:

just a layering issue. the wheel does not appear in front of you.

### Round ID:

[246289](https://scrubby.melonmesa.com/round/246289)

### Test Merge Information:

- [Make the xmas period longer](https://www.github.com/tgstation/tgstation/pull/88581)
- [Pet Commands QOL . makes pet commands easier to use](https://www.github.com/tgstation/tgstation/pull/88495)

### Reproduction Steps:

sit on a wheelchair

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

Closes #88644
Insanity

## Changelog
:cl:
fix: Fixed chair, echair, wheelchair and vehicle overlays on painted objects
/:cl:

### CODE REVIEW

File: code/game/objects/structures/beds_chairs/chair.dm [hunk 1]
```diff
@@ -77,28 +77,44 @@
...
+/obj/structure/chair/update_atom_colour()
+	. = ..()
+	if (armrest)
+		color_atom_overlay(armrest)
+
+/obj/structure/chair/color_atom_overlay(mutable_appearance/overlay)
+	overlay.color = color
+	if (!cached_color_filter)
+		return overlay
+	to_chat(world, "[overlay.icon_state]")
```
Comment (00-Steven) [inline code/game/objects/structures/beds_chairs/chair.dm:89]: hey uhhh is this supposed to still be in here

Comment (SmArtKar) [reply code/game/objects/structures/beds_chairs/chair.dm:89]: :)

File: code/datums/components/electrified_buckle.dm [hunk 2]
```diff
@@ -153,8 +153,9 @@
 /datum/component/electrified_buckle/proc/on_update_overlays(atom/movable/source, list/overlays)
 	SIGNAL_HANDLER
 	var/overlay_layer = length(source.buckled_mobs) ? ABOVE_MOB_LAYER : OBJ_LAYER
-	for (var/image/overlay_image in requested_overlays)
+	for (var/mutable_appearance/overlay_image in requested_overlays)
```
Comment (LemonInTheDark) [inline code/datums/components/electrified_buckle.dm:156]: name it something else

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/datums/components/electrified_buckle.dm [hunk 3]
```diff
@@ -153,9 +153,10 @@
 /datum/component/electrified_buckle/proc/on_update_overlays(atom/movable/source, list/overlays)
 	SIGNAL_HANDLER
 	var/overlay_layer = length(source.buckled_mobs) ? ABOVE_MOB_LAYER : OBJ_LAYER
-	for (var/image/overlay_image in requested_overlays)
-		overlay_image.layer = overlay_layer
-		overlays += overlay_image
+	for (var/mutable_appearance/electrified_overlay as anything in requested_overlays)
```
Comment (LemonInTheDark) [inline code/datums/components/electrified_buckle.dm:156]: is an as anything here safe? (text?) maybe an if(!(isimage && isma()) guard check?
also TECHNICALLY risks issues with static MAs but like I don't care

Comment (LemonInTheDark) [reply code/datums/components/electrified_buckle.dm:156]: if you wanna be safe do a copy first

**Pull request merged**